### ISSUE TITLE
Search icon at the website does not work?

### ISSUE THREAD

**Issue by Conxz** [NeuroVault/NeuroVault#531]
Hi~
I find that the Search icon does not work at the website,  with either Chrome, IE or Edge. I'm not sure whether it's related to the network connection in China, but other functions work well. 

http://www.neurovault.org/search

Is there any other way to search for a map with keywords? 

![neurovault](https://cloud.githubusercontent.com/assets/712044/16516232/730a7258-3fa9-11e6-9365-71d7450eb575.jpg)

**Comment by erramuzpe** [comment]
Hi @Conxz , 

the search engine is powered by Google's custom search, so it could be due to a region-restriction.

You can try to search maps by tag: http://www.neurovault.org/images/tags/{your_query_tag}

or search for collections from collections view: http://www.neurovault.org/collections/?q={your_query}

**Comment by chrisfilo** [comment]
Is there a Baidu version of custom search engine we could use instead for
requests coming from China?

**Comment by erramuzpe** [comment]
Indeed, there is. There are tutorials to get html snippets [here](http://developer.baidu.com/#baidu) and [here](http://www.dragonmetrics.com/how-to-optimize-your-site-with-baidu-webmaster-tools/). Is this what we are looking for?

> Baidu Search Box
> 
> Enable Baidu custom search box in your site pages by copy & pasting the HTML code from Baidu Webmaster Tools. Learn more about getting Baidu Search Box code here.

**Comment by chrisfilo** [comment]
It seems it requires a Chinese phone number to set up (and knowledge of the
language). It would be nice to provide the search feature to Chinese users
but we would need an insider to help us. @Conxz maybe you could take this
on?

**Comment by Conxz** [comment]
Thanks a lot. These two links work well. 
http://www.neurovault.org/images/tags/
http://www.neurovault.org/collections/?q=
On both pages, the search input field works. 

As I understand, the Search icon in NeuroVault is designed for site search. This is different from the two links above. Is this correct? 
Here is the link for the site search function.
http://zhanzhang.baidu.com/guide/index
@chrisfilo No problem. Hope I can help.

**Comment by Conxz** [comment]
Hi all,
After some tries, I find that there are two ways to implement the search feature with Baidu. 
1. The first one does not need site validation. I've just finished a demo [here](http://www.conxz.net/blog/tools/bsearch), so you can play with it. The bad thing is that the search results are shown in a new page. 
2. The second one needs a site validation. In this way, Baidu provides a page for searching within the site of interest and showing results. To finish the site validation, we need to put the following line between <head> and </head>.
<meta name="baidu-site-verification" content="69JDAzomtM" />

**The styles of both ways are different from the Google one** (I find that the google search works well for me with **https** protocol, i.e., gcse.src = 'https://cse.google.com/cse.js?cx=' + cx). This is an issue for maintaining consistency.
In addition, I have to say that **Baidu does poor in English searching**. 

So here is my opinion about this issue. Redirect the Search icon to [the ‘all collections’ page](http://www.neurovault.org/collections/?q=)?. If we try the second approach, the site validation is needed.

**Comment by erramuzpe** [comment]
Hey,
I've been trying some queries on your demo (cool btw), and indeed it's true that Baidu results are not very good.  The problem with redirecting is that it will only search over collections, while the Google Search does it over collections, images, tasks... 
We could think in something intermediate, say a warning text field or something similar ("if the search engine is not working for you, try the search in collections/tag page")

**Comment by Conxz** [comment]
I got it. This would be better. 
In addition, as emntioned above, the Google search with https ptotocol works well for me (via editing host file as usual). Given that Google gives the https link for site search by default, I wonder what is the consideration of using `gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; ` for NeuroVault, instend of `gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;` as given by default?
